metrics/sources/summary/summary.go

```diff
@@ -194,10 +194,22 @@ func (this *summaryMetricsSource) decodePodStats(metrics map[string]*MetricSet,
 
 	for _, container := range pod.Containers {
 		key := PodContainerKey(ref.Namespace, ref.Name, container.Name)
+		if _, exist := metrics[key]; exist && containerIsTerminated(&container, metrics[key].CreateTime) {
+			continue
+		}
 		metrics[key] = this.decodeContainerStats(podMetrics.Labels, &container, false)
 	}
 }
 
+func containerIsTerminated(container *stats.ContainerStats, otherStartTime time.Time) bool {
+	if container.StartTime.Time.Before(otherStartTime) && *container.CPU.UsageNanoCores == 0 {
```

Comment:
How about avoiding double checking of the same condition `container.StartTime.Time.Before(otherStartTime)` by implementing this in the following way:
```
if container.StartTime.Time.Before(otherStartTime) {
    if *container.CPU.UsageNanoCores == 0 {
        return true
    }
    glog.Warningf(...)
}
```